lib/srv/db/proxyserver.go

```diff
@@ -583,9 +583,13 @@ func (s *ProxyServer) getConfigForServer(ctx context.Context, identity tlsca.Ide
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+
 	response, err := s.cfg.AuthClient.SignDatabaseCSR(ctx, &proto.DatabaseCSRRequest{
 		CSR:         csr,
 		ClusterName: identity.RouteToCluster,
+		// Field SignWithDatabaseCA was introduced in v9.1. Old clients don't send it
+		// which will resolve as false on the other side.
+		SignWithDatabaseCA: true,
```

Comment:
@jakule 
Not sure if this filed should be always set to true. 

This is the proxy flow where proxy asks the auth service to issue a certifcates that will be used to establish a connection to db agent. 

Let's consider following case: 

1) Proxy and auth services are running old version so there is only `HostCA`
2) After teleport updated the `DatabaseCA` was created by migration and `DatabaseCA == HostCA`
3) There is a legacy DB agent that terminates DB reverse tunnel connections with `HostCA` thus `DatabaseCA == HostCA` the connection established by an updated proxy with Cert created by DatabaseCA works.
4) `HostCA` was rotated. 
5) Thus `HostCA == DatabaseCA` the `DatabaseCA` also needs to be rotated. In the result HostCA != DatabaseCA
6) Legacy DB agent still uses `HostCA` for validation of incoming connections.
7) The updated teleport proxy wants to establish a connection to the db agent so in the updated version the `SignWithDatabaseCA` is always true so proxy request service auth to issue  cert to connect to legacy DB agent but there cert will be signed by `DatabaseCA` where the legacy DB agent is still using `HostCA` Thus in step 4) the `HostCA` was rotated and `HostCA != DatabaseCA` the connection cannot be established.